src/frontend/hooks/useLibrary.ts

```diff
@@ -0,0 +1,71 @@
+import { Category } from 'frontend/types'
+import React from 'react'
+
+import { GameInfo, InstalledInfo } from 'common/types'
+import {
+  gogLibraryStore,
+  gogInstalledGamesStore,
+  sideloadLibrary,
+  libraryStore
+} from 'frontend/helpers/electronStores'
+
+type Props = {
+  category: Category
+}
+
+const useLibrary = ({ category }: Props): Array<GameInfo> => {
+  const [library, setLibrary] = React.useState<Array<GameInfo>>([])
+  const [gogLibrary, setGogLibrary] = React.useState<Array<GameInfo>>([])
+  const [epicLibrary, setEpicLibrary] = React.useState<Array<GameInfo>>([])
+  const [sideloadedLibrary, setSideloadedLibrary] = React.useState<
+    Array<GameInfo>
+  >([])
+
+  const getLibrary = (): Array<GameInfo> => {
+    const games = gogLibraryStore.get('games', []) as GameInfo[]
+    const installedGames = gogInstalledGamesStore.get(
+      'installed',
+      []
+    ) as Array<InstalledInfo>
+    for (const igame in games) {
+      for (const installedGame of installedGames) {
+        if (installedGame.appName === games[igame].app_name) {
+          games[igame].install = installedGame
+          games[igame].is_installed = true
+        }
+      }
+    }
+
+    return games
+  }
+
+  React.useEffect(() => {
+    setGogLibrary(getLibrary())
+    setEpicLibrary(libraryStore.get('library', []) as GameInfo[])
+    setSideloadedLibrary(sideloadLibrary.get('games', []) as GameInfo[])
+  }, [])
+
+  React.useEffect(() => {
+    switch (category) {
+      case 'all':
+        setLibrary([...gogLibrary, ...epicLibrary, ...sideloadedLibrary])
+        break
+      case 'gog':
+        setLibrary(gogLibrary)
+        break
+      case 'legendary':
+        setLibrary(epicLibrary)
+        break
+      case 'sideload':
+        setLibrary(sideloadedLibrary)
+        break
+      default:
+        setLibrary([])
```

Comment:
should we fallback to all the games instead of an empty array? I mean, if something goes wrong we would show no games, and we have had issues with an empty least because of some issue with the category.

I agree that it's more correct to return `[]`, but maybe it's safer to fallback to all on something unexpected